psl-clojure/template/resources/leiningen/new/psl_clojure/core.clj

```diff
@@ -0,0 +1,171 @@
+                  (psl/NEQ 'P1 'P2))
+                 (knows 'P1 'P2))
+                20 true "nearby-people-known")
+  (psl/add-rule model
+                (psl/IMPL
+                 (psl/AND
+                  (lived 'P1 'L1)
+                  (lived 'P2 'L2)
+                  (psl/NEQ 'P1 'P2)
+                  (psl/NEQ 'L1 'L2))
+                 (psl/NOT (knows 'P1 'P2)))
+                5 true "distant-people-unknown")
+  (psl/add-rule model
+                (psl/IMPL
+                 (psl/AND
+                  (likes 'P1 'T)
+                  (likes 'P2 'T)
+                  (psl/NEQ 'P1 'P2))
+                 (knows 'P1 'P2))
+                10 true "homophily")
+  (psl/add-rule model
+                (psl/IMPL
+                 (psl/AND
+                  (knows 'P1 'P2)
+                  (knows 'P2 'P3)
+                  (psl/NEQ 'P1 'P3))
+                 (knows 'P1 'P3))
+                5 true "transitive")
+  (psl/add-rule-string model datastore "knows(P1, P2) = knows(P2, P1) .")
+  (psl/add-rule model
+                (psl/NOT (knows 'P1 'P2))
+                5 true "prior")
+  model)
+
+;;; Handy functions --- modify to suit
+
+(defn config-bundle
+  "A config bundle"
+  ([]
+   (config-bundle "psl"))
+  ([bundle-name]
+   (let [cm (. org.linqs.psl.config.ConfigManager getManager)]
+     (.getBundle cm bundle-name))))
+
+(defn data-store
+  "A data store"
+  ([config-bundle]
+   (data-store config-bundle true "."))
+  ([config-bundle clear-db output-dir]
+   (let [db-path (.getPath (io/file output-dir "psl"))
+         driver (H2DatabaseDriver. H2DatabaseDriver$Type/Disk db-path clear-db)]
+     (org.linqs.psl.database.rdbms.RDBMSDataStore. driver config-bundle))))
+
+(defn inference
+  "Run inference"
+  [cb ds m obs res closed-preds]
+  (let [;; A view for reading observations and writing inferences
+        db (psl/open-db ds m [obs] res closed-preds)
+
+        ;; An inference application
+        inf (psl/default-inference m db cb)]
+
+    ;; Run inference
+    (psl/mpe-inference inf)
+
+    ;; Retrieve ground rules
+    (def grs (doall (for [gr (.getGroundRules (.getGroundRuleStore inf))] gr)))
+
+    ;; Clean up
+    (psl/close-db db)
+    (.close inf)))
+
+(defn evaluate
+  "Evaluate inference results"
+  [datastore model target-partition truth-partition closed-preds open-pred]
+  (let [evaluator (ContinuousEvaluator.)
+        rv-db (let [dummy-write (psl/partition-new datastore)]
+                (psl/open-db datastore model [target-partition] dummy-write closed-preds))
+        truth-db (let [dummy-write (psl/partition-new datastore)]
+                   (psl/open-db datastore model [truth-partition] dummy-write closed-preds))
+        open-pred (psl/p model open-pred)]
+    (.compute evaluator rv-db truth-db open-pred)
+    (prn (.toString (.getAllStats evaluator)))
+    (psl/close-db rv-db)
+    (psl/close-db truth-db)))
+
+(defn example []
+  "Run the full example"
+  (def cb (config-bundle))
```

Comment:
I don't like these short names, they are not informative.
I believe the current examples use more complete names.